lib/src/component_declaration/builder_helpers.dart

```diff
@@ -134,6 +134,91 @@ abstract class UiProps extends component_base.UiProps with GeneratedClass {
   @toBeGenerated PropsMetaCollection get staticMeta => throw UngeneratedError(member: #meta);
 }
 
+
+/*
+UiComponent:
+..addProps(copyUnconsumedProps())
+
+UiComponent2:
+..modifyProps(addUnconsumedProps)
+
+UiFunction and UiComponent2:
+final consumedProps = props.staticMeta.forMixins({BarPropsMixin});
+...
+..addUnconsumedProps(props, consumedProps)
+
+
+Proposed:
+// By default the method not return props for the class it is being called on which makes it super easy and convenient.
+// eg. if `props` is of type `MyPropsMixin` then it would be equivalent to something like `props.getUnconsumed(consumed: {MyPropsMixin})`
+
+
+..addAll(props.getUnconsumed())
+
+..addAll(props.getUnconsumed({MyProps}));
+..addAll(props.getUnconsumed(consumed: {MyProps}));
+..addAll(props.getUnconsumedProps(consumedPropsMixins: {MyProps}))
+
+..addAll(props.getUnconsumed(exclude: {MyProps}))
+
+*/
+
+extension UnusedProps<T extends UiProps> on T {
+
+  /// Returns this instance's props map excluding the keys found in [consumedMixins].
+  ///
+  /// [consumedMixins] should be a `Set` of PropsMixin `Type`s.
+  /// If [consumedMixins] is not set it defaults to using the current instances Type.
+  ///
+  /// __Example:__
+  ///
+  /// ```dart
+  /// // within a functional component: `uiFunction<FooPropsMixin>`
+  /// // filter out the current components props when forwarding to Bar.
+  /// return (Bar()..addAll(props.getConsumed()))();
+  /// ```
+  /// OR
+  /// ```dart
+  /// // within a functional component that has multiple mixins on a Props class: `uiFunction<FooProps>`
+  /// // filter out the Components props when forwarding to Bar.
+  /// return (Bar()..addAll(props.getConsumed(consumedMixins:{FooPropsMixin}))();
+  /// ```
+  ///
+  /// To only add DOM props, use [addUnconsumedDomProps].
+  ///
+  /// Related: `UiComponent2`'s `addUnconsumedProps`
+  Map getPropsToForward({Set<Type> exclude, Set<Type> include, bool domOnly = false}) {
+    final useDefaultExclude = exclude == null && (include == null || (include is Set && include.isEmpty));
+    try {
+      Iterable<PropsMeta> includedProps = [];
+      final unconsumedProps = {};
```

Comment:
Should we make a form of this function that uses modifyProps and passes it to propsToUpdate so we're not making an unnecessary copy?